syslog-ng-ctl/syslog-ng-ctl.c

```diff
@@ -45,13 +45,13 @@ typedef struct _SubCommandDescriptor
   const gchar *mode;
   const GOptionEntry *options;
   const gchar *description;
-  gint (*main)(gint argc, gchar *argv[], const gchar *mode);
+  gint (*main)(gint argc, gchar *argv[], const gchar *mode, GOptionContext *ctx);
```

Comment:
What if we pass just the string instead of `GOptionContext`?
That case signature would be more understandable for other developers, IMHO.